### ISSUE TITLE
Pipeline Metrics not appearing on UI

### ISSUE THREAD

**Issue by cai-harry** [kubeflow/pipelines#4763]
### What steps did you take:
I am building a pipeline using `dslContainerOp`. I have a component called `traineval` which is supposed to output pipeline metadata (tensorboard) and metrics (accuracy score).

When defining my pipeline:
```
    train_eval = dsl.ContainerOp(
        name='Train_and_Evaluate',
        image=<image name>,
        arguments=[
            google_project,
            preprocessed_texts,
            preprocessed_labels,
            model_uri,
            tensorboard_dir,
            num_epochs
        ],
        file_outputs={
            'mlpipeline-metrics': '/mlpipeline-metrics.json',
            'mlpipeline-ui-metadata': '/mlpipeline-metadata.json',
            'model': '/model.h5'
        }
    )
    train_eval.container.set_image_pull_policy('Always')
    train_eval.after(pre_process)
```

And inside `traineval`, I'm saving the accuracy score as a local json file:
```
def output_metrics(accuracy):
    metrics = {
        'metrics': [
            {
                'name': 'accuracy-score',
                'numberValue': accuracy,
                'format': "PERCENTAGE",
            }
        ]
    }
    with file_io.FileIO('/mlpipeline-metrics.json', 'w') as f:
        json.dump(metrics, f)
```

### What happened:
The `file_output`s that I defined do appear under output artifacts:
![image](https://user-images.githubusercontent.com/35222594/98938301-91f52080-24df-11eb-8906-a9e0f9b45433.png)

And `mlpipeline-ui-metadata` is handled correctly (a tensorboard artifact is generated), but the `mlpipeline-metrics` does not appear in the UI.

![image](https://user-images.githubusercontent.com/35222594/98934739-73d8f180-24da-11eb-9c78-7875f4ea050a.png)

The metrics don't appear as a column in the list of runs either.

![image](https://user-images.githubusercontent.com/35222594/98934851-9539dd80-24da-11eb-9dda-484bbd078bb4.png)

### What did you expect to happen:

I expect my accuracy score to appear where it currently says "No metrics found for this run."

I also expect the accuracy score to appear as a column in the list of runs, similar to your documentation.

![image](https://www.kubeflow.org/docs/images/taxi-tip-run-scores.png)

### Environment:

How did you deploy Kubeflow Pipelines (KFP)?
Deployed from manifests

KFP version: v1.0.0 988f5b02e4211dfff1c02eb0b9a52cbc69793364

KFP SDK version:
```
kfp                      1.0.1
kfp-server-api           1.0.3
```

### Anything else you would like to add:
This is a similar issue to #3809 but unlike in that issue, I have named everything correctly and it still doesn't work.

/kind bug

**Comment by Bobgy** [comment]
Hi @cai-harry, sorry for the issue. You would want to look at the updated documentation not merged yet: https://github.com/kubeflow/website/pull/2171.

**Comment by cai-harry** [comment]
Sorry, I don't understand what changes I actually need to make? The updated documentation seems to use the `create_component_from_func` approach rather than the `dsl.ContainerOp` approach?

**Comment by Bobgy** [comment]
@cai-harry container op has been deprecated, you won't be able to use it for metrics

**Comment by cai-harry** [comment]
For the benefit of anyone looking at this thread: I have changed to kubeflow v1.0.4 and it is not an issue anymore.
![image](https://user-images.githubusercontent.com/35222594/105633609-8617ca00-5e51-11eb-9948-7120c0bdc38a.png)

**Comment by tuvovan** [comment]
can you share the code snippet to make the tensorboard shown?

**Comment by swamat** [comment]
@cai-harry Can you please help me in fixing this issue. I am using v1.0.4. STill am not able to see the metrics in UI
